Apps/phzoom/zoom_connector.py

```diff
@@ -0,0 +1,513 @@
+
+        if phantom.is_fail(ret_val):
+            self.save_progress("Test Connectivity Failed")
+            return action_result.get_status()
+
+        self.save_progress("Test Connectivity Passed")
+        self.save_progress("Connected to Zoom API successfully")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_get_user_settings(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        user_id = param['user_id']
+
+        ret_val, response = self._make_rest_call('/users/{}/settings'.format(user_id), action_result, params=None, headers=None)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(response)
+        return action_result.set_status(phantom.APP_SUCCESS, 'User settings for id {} successfully retrieved'.format(user_id))
+
+    def _handle_update_user_settings(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        user_id = param['user_id']
+
+        pmi_password = self._get_password(param.get('pmi_password'), param.get('gen_pmi_password'))
+        waiting_room = param.get('waiting_room')
+        req_password_sched = param.get('req_password_sched')
+        req_password_inst = param.get('req_password_inst')
+        req_password_pmi = param.get('req_password_pmi')
+
+        if not(pmi_password or waiting_room != "None" or req_password_sched != "None" or req_password_inst != "None"):
+            return action_result.set_status(phantom.APP_ERROR, 'No settings were selected for update')
+
+        data = {}
+
+        if pmi_password or req_password_sched != 'None' or req_password_inst != 'None' or req_password_pmi != 'None':
+            data['schedule_meeting'] = {}
+            if pmi_password:
+                data['schedule_meeting']['pmi_password'] = pmi_password
+            if req_password_sched:
+                data['schedule_meeting']['require_password_for_scheduling_new_meetings'] = req_password_sched == 'True'
+            if req_password_inst:
+                data['schedule_meeting']['require_password_for_instant_meetings'] = req_password_inst == 'True'
+            if req_password_pmi:
+                data['schedule_meeting']['require_password_for_pmi_meetings'] = ('all' if req_password_pmi == 'True' else 'none')
+        if waiting_room != 'None':
+            data['in_meeting'] = {'waiting_room': waiting_room == 'True'}
+
+        ret_val, _ = self._make_rest_call('/users/{}/settings'.format(user_id), action_result, json=data, headers=None, method="patch")
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.update_summary({
```

Comment:
There are no example summaries and messages in the json

Reply:
Fixed